app/controllers/api/base_controller/renderer.rb

```diff
@@ -45,6 +45,7 @@ def collection_to_jbuilder(type, reftype, resources, opts = {})
 
           unless @req.hide?("resources") || collection_option?(:hide_resources)
             key_id = collection_config.resource_identifier(type)
+            opts[:collection_search] = true
```

Comment:
* Hmm? It's not intuitive to me that this block of code directly relates to `collection_search` 😕  I probably just don't understand the implications of the logic on line 46
* Even if 'we don't want to expand custom actions on collection search' is your intention, don't you just want to pass an option for `expand_custom_actions` in this method to maintain clean semantics that are already present in `resource_to_jbuilder` instead of teaching `resource_to_jbuilder` that there's such a thing as a collection search that it seems to care about for some reason?

Reply:
both good 👍